burnman/eos/birch_murnaghan_4th.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import absolute_import
+# This file is part of BurnMan - a thermoelastic and thermodynamic toolkit for the Earth and Planetary Sciences
+# Copyright (C) 2012 - 2015 by the BurnMan team, released under the GNU GPL v2 or later.
+
+
+import scipy.optimize as opt
+from . import equation_of_state as eos
+from ..tools import bracket
+import warnings
+
+def bulk_modulus_fourth(volume, params):
+    """
+    compute the bulk modulus as per the third order
+    birch-murnaghan equation of state.  Returns bulk
+    modulus in the same units as the reference bulk
+    modulus.  Pressure must be in :math:`[Pa]`.
+    """
+    B1 = (params['K_0']*params['Kprime_prime_0'])+((params['Kprime_0']-4.)*(params['Kprime_0']-5.))+(59./9.)
+    B2 = (3.*params['K_0']*params['Kprime_prime_0'])+((params['Kprime_0']-4.)*((3.*params['Kprime_0'])-13.))+(129./9.)
+    B3 = (3.*params['K_0']*params['Kprime_prime_0'])+((params['Kprime_0']-4.)*((3.*params['Kprime_0'])-11.))+(105./9.)
+    B4 = (params['K_0']*params['Kprime_prime_0'])+((params['Kprime_0']-4.)*(params['Kprime_0']-3.))+(35./9.)
+
+    x = params['V_0']/volume
+
+    K = (9./16.)*params['K_0']*( (-5.*B1/3.)*pow(x,-5./3.)+(7.*B2/3.)*pow(x,-7./3.)-3*B3*pow(x,-3.)+(11.*B4/3.)*pow(x,-11./3.))
+    return K
+
+def volume(pressure, params):
```

Comment:
I was referring to the volume function, not the bulk modulus function. You
do call the bulk modulus function down below, so you need it!

On Tue, Jan 5, 2016 at 11:09 AM, Cayman Unterborn <notifications@github.com>
wrote:

> In burnman/eos/birch_murnaghan_4th.py
> <https://github.com/geodynamics/burnman/pull/158#discussion_r48882901>:
>
> > +    compute the bulk modulus as per the third order
> > +    birch-murnaghan equation of state.  Returns bulk
> > +    modulus in the same units as the reference bulk
> > +    modulus.  Pressure must be in :math:`[Pa]`.
> > +    """
> > +    B1 = (params['K_0']*params['Kprime_prime_0'])+((params['Kprime_0']-4.)*(params['Kprime_0']-5.))+(59./9.)
> > +    B2 = (3.*params['K_0']*params['Kprime_prime_0'])+((params['Kprime_0']-4.)*((3.*params['Kprime_0'])-13.))+(129./9.)
> > +    B3 = (3.*params['K_0']*params['Kprime_prime_0'])+((params['Kprime_0']-4.)*((3.*params['Kprime_0'])-11.))+(105./9.)
> > +    B4 = (params['K_0']*params['Kprime_prime_0'])+((params['Kprime_0']-4.)*(params['Kprime_0']-3.))+(35./9.)
> > +
> > +    x = params['V_0']/volume
> > +
> > +    K = (9./16.)*params['K_0']*( (-5.*B1/3.)*pow(x,-5./3.)+(7.*B2/3.)*pow(x,-7./3.)-3*B3*pow(x,-3.)+(11.*B4/3.)*pow(x,-11./3.))
> > +    return K
> > +
> > +def volume(pressure, params):
>
> I don't call it. I believe when I was making BM4 the paper I was pulling
> from had both so I just made something for K as well.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/burnman/pull/158/files#r48882901>.
>